### ISSUE TITLE
Removing return not sugggested by `needless_return` when calling some chained string methods

### ISSUE THREAD

**Issue by cyrgani** [rust-lang/rust-clippy#12907]
### Summary

When having a return statement with a call to `str::trim`, the `needless_return` list does not warn. Replacing `trim` with`len` in the following example correctly warns.

It seems to happen with other borrowing methods as well, such as in the following example:
```rust
fn f() -> String {
	return "".split("").next().unwrap().to_string();
}
```

### Lint Name

needless_return

### Reproducer

I tried this code:

```rust
fn main(){
	f();
}
fn f() -> String {
	return "".trim().to_string();
}
```

I expected to see this happen:
```
warning: unneeded `return` statement
 --> src/main.rs:5:2
  |
5 |     return "".trim().to_string();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
  = note: `#[warn(clippy::needless_return)]` on by default
help: remove `return`
  |
5 -     return "".trim().to_string();
5 +     "".trim().to_string()
  |
```

Instead, this happened:

No output

### Version

```text
rustc 1.78.0 (9b00956e5 2024-04-29)
binary: rustc
commit-hash: 9b00956e56009bab2aa15d7bff10916599e3d6d6
commit-date: 2024-04-29
host: x86_64-unknown-linux-gnu
release: 1.78.0
LLVM version: 18.1.2
```

**Comment by y21** [comment]
We do need to be careful here because allowing these cases can sometimes also end up as false positives when temporaries are involved, e.g.:
```rs
fn f() -> i32 {
  let r = RefCell::new(42);
  return *r.borrow();
}
```
Trying to replace that `return <expr>;` with simply `<expr>` results in a compile error because the temporary that references the local `RefCell` gets dropped after the local itself
